test/e2e/storage/regional_pd.go

```diff
@@ -0,0 +1,346 @@
+		framework.ExpectNoError(c.StorageV1().StorageClasses().Delete(class.Name, nil),
+			"Error deleting StorageClass %s", class.Name)
+	}()
+
+	By("creating a StatefulSet")
+	_, err = c.CoreV1().Services(ns).Create(service)
+	Expect(err).NotTo(HaveOccurred())
+	_, err = c.AppsV1().StatefulSets(ns).Create(statefulSet)
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.WaitForStatefulSetReplicasReady(statefulSet.Name, ns, c, framework.Poll, framework.PodStartTimeout)
+
+	pvc := getPVC(c, ns, repdLabels)
+	defer func() {
+		framework.Logf("deleting statefulset%q/%q", statefulSet.Namespace, statefulSet.Name)
+		// typically this claim has already been deleted
+		framework.ExpectNoError(c.AppsV1().StatefulSets(ns).Delete(statefulSet.Name, nil /* options */),
+			"Error deleting StatefulSet %s", statefulSet.Name)
+		framework.Logf("deleting claim %q/%q", pvc.Namespace, pvc.Name)
+		framework.ExpectNoError(c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(pvc.Name, nil),
+			"Error deleting claim %s.", pvc.Name)
+		err = framework.WaitForPersistentVolumeDeleted(c, pvc.Spec.VolumeName, framework.Poll, pvDeletionTimeout)
+		if err != nil {
+			framework.Logf("WARNING: PV %s is not yet deleted, and subsequent tests may be affected.", pvc.Spec.VolumeName)
+		}
+	}()
+
+	By("getting zone information from pod")
+	pod := getPod(c, ns, repdLabels)
+	nodeName := pod.Spec.NodeName
+	node, err := c.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	podZone := node.Labels[apis.LabelZoneFailureDomain]
+
+	// TODO (verult) Consider using node taints to simulate zonal failure instead.
+	By("deleting instance group belonging to pod's zone")
+	cloud, err := framework.GetGCECloud()
+	if err != nil {
+		Expect(err).NotTo(HaveOccurred())
+	}
+	instanceGroupName := framework.TestContext.CloudConfig.NodeInstanceGroup
+	instanceGroup, err := cloud.GetInstanceGroup(instanceGroupName, podZone)
+	Expect(err).NotTo(HaveOccurred(),
+		"Error getting instance group %s in zone %s", instanceGroupName, podZone)
+	err = framework.DeleteManagedInstanceGroup(podZone)
+	Expect(err).NotTo(HaveOccurred(),
+		"Error deleting instance group in zone %s", podZone)
+	defer func() {
+		framework.Logf("recreating instance group %s", instanceGroup.Name)
+
+		// HACK improve this when Managed Instance Groups are available through the cloud provider API
+		templateName := strings.Replace(instanceGroupName, "group", "template", 1 /* n */)
+
+		framework.ExpectNoError(framework.CreateManagedInstanceGroup(instanceGroup.Size, podZone, templateName),
+			"Error recreating instance group %s in zone %s", instanceGroup.Name, podZone)
+		framework.ExpectNoError(framework.WaitForReadyNodes(c, nodeCount, framework.RestartNodeReadyAgainTimeout),
+			"Error waiting for nodes from the new instance group to become ready.")
+	}()
+
+	// Rescheduling of the pod should occur while instance group deletion is in progress.
```

Comment:
Yeah I would wait and validate that the pod has restarted to avoid possible race conditions.